lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -112,6 +112,11 @@ def add_auth_purchase(post, money, payment, options)
         add_ewallet(post, options)
         add_payment_fields(post, options)
         add_payment_urls(post, options)
+        add_idempotency(options)
+      end
+
+      def add_idempotency(options)
+        @options[:idempotency] = options[:idempotency] if options[:idempotency]
```

Comment:
Looks like the if statement is not needed

Reply:
why idempotency is an instance variable?